### ISSUE TITLE
Crash after all the photos are taken and is happen onfido dismiss screen

### ISSUE THREAD

**Issue by tnmendes** [onfido/onfido-ios-sdk#60]
### What was the expected behaviour?

After taking all the documents photos it's expected that Onfido screen dismiss smoothly to my UIViewController

### What happened instead?

The dismiss happen and at the same time crash occurs.

### Version info:

- Onfido SDK (Debug/Release): Onfido 10.1.0 Debug
- iOS: 12
- Xcode: 10.1
- Device/Simulator: device iPhone 6s
- Device/Simulator language: English

### Steps to reproduce:

If I open Onfido by using my configuration  ".withDocumentStep(ofType: documentType, andCountryCode: coutryCode)"
this is the error that I get:

```
Swinject: Resolution failed. Expected registration:
	{ Service: NavigationControllerProxy, Factory: (Resolver) -> NavigationControllerProxy }
Available registrations:
```

With ".withDocumentStep()" this is the errro:

```
Swinject: Resolution failed. Expected registration:
	{ Service: NavigationControllerProxy, Factory: (Resolver) -> NavigationControllerProxy }
Available registrations:
Fatal error: Unexpectedly found nil while unwrapping an Optional value
2018-12-12 09:41:02.301205+0000 ios_assure[2618:1255954] Fatal error: Unexpectedly found nil while unwrapping an Optional value
```

### Integration configuration:

I am not using storyboards, I am using programmatic interfaces (Snapkit)

```
//
//  OnfidoViewController.swift

import UIKit
import Onfido
import Alamofire


class OnfidoViewController: GenericViewController<OnfidoView> {
    
    var metadata: Metadata
    
    init(metadata: Metadata) {
        
        self.metadata = metadata
        super.init()
    }
    
    
    public required init?(coder: NSCoder){
        
        self.metadata = Metadata(url: URL(string: "")!)!
        super.init(coder: coder)
    }
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        contentView.customBar.buttonLeft.addTarget(self, action: #selector(self.goBack), for: .touchUpInside)
        contentView.button.addTarget(self, action: #selector(self.tryAgainPressed), for: .touchUpInside)
        self.initFlow()
    }
    
    
    @objc private func goBack() {
        
        self.dismiss(animated: true, completion: {})
    }
    
    @objc private func tryAgainPressed() {
        self.initFlow()
    }
    
    
    private func initFlow() {
        
        let helper = OnfidoHelper()
        helper.createApplicant{ (applicantId, error) in
            
            guard error == nil else {
                self.showErrorMessage(forError: error!)
                return
            }
            
            self.runFlow(forApplicantWithId: applicantId!)
        }
    }
    
    
    private func responseHandler(response: OnfidoResponse) {
        
        if case let OnfidoResponse.error(innerError) = response {
            self.showErrorMessage(forError: innerError)
            self.contentView.label.text = "Something happened"
            self.contentView.button.isHidden = true
        } else if case OnfidoResponse.success = response {
            showAlertMessage(title: "Success", message: "Success")
            self.contentView.label.text = "Success"
            self.contentView.button.isHidden = false
        } else if case OnfidoResponse.cancel = response {
            showAlertMessage(title: "Canceled", message: "Canceled by user")
            self.contentView.label.text = "Canceled"
            self.contentView.button.isHidden = false
        }
    }
    
    
    private func errorHandle(onfidoError: OnfidoFlowError){
        
        var text = "Something happened"
        switch onfidoError {
        case OnfidoFlowError.cameraPermission: text = ""; break
        case OnfidoFlowError.failedToWriteToDisk: break
        case OnfidoFlowError.microphonePermission: break
        default: self.showErrorMessage(forError: onfidoError)
        }
        
        self.contentView.label.text = text
        self.contentView.button.isHidden = true
    }
    
    
    private func runFlow(forApplicantWithId applicantId: String) {
        
        let responseHandler: (OnfidoResponse) -> Void = { response in
            self.responseHandler(response: response)
        }
        
        guard let coutryCode = metadata.coutryType?.rawValue else { return }
        guard let documentType = metadata.documentType?.getOnfidoType()  else { return }
        
        let config = try! OnfidoConfig.builder()
            .withToken(Configuration.kOnfidoMobileToken)
            .withApplicantId(applicantId)
            //.withDocumentStep()
            //.withFaceStep(ofVariant: .photo)
            .withAppearance(contentView.getOnfidoAppearance()) // UI customisation
            .withDocumentStep(ofType: documentType, andCountryCode: coutryCode) // Flow customisation, documents select
            .build()
        
        let onfidoFlow = OnfidoFlow(withConfiguration: config)
            .with(responseHandler: responseHandler)
        
        do {
            
            let onfidoRun = try onfidoFlow.run()
            onfidoRun.modalPresentationStyle = .formSheet // to present modally
            self.present(onfidoRun, animated: true, completion: {})
            
        } catch let error {
            
            // cannot execute the flow
            // check CameraPermissions
            self.showErrorMessage(forError: error)
        }
    }
    
    
    private func showErrorMessage(forError error: Error) {
        
        showAlertError(message: "Onfido SDK Errored \(error)")
    }
    
}


enum ApplicantError: Error {
    case apiError([String:Any])
}


public func getOnfidoAppearance() -> Appearance {
        
        let appearance = Appearance(
            primaryColor: UIColor.red,
            primaryTitleColor: UIColor.green,
            primaryBackgroundPressedColor: UIColor.gray,
            secondaryBackgroundPressedColor: UIColor.purple)
        return appearance
    }
```

**Comment by anuragajwani** [comment]
Hi @tnmendes ,

Sorry to hear that you are having issues. I have tested out your config in a new project and am unable to reproduce your issue. I was unable to get hold of an iPhone 6s but tested it on various other devices as well as simulator:

```
//
//  ViewController.swift
//  Onfido-10-1-0-issue-60
//
//  Created by Anurag Ajwani on 12/12/2018.
//  Copyright © 2018 Onfido. All rights reserved.
//

import UIKit
import Onfido
import Alamofire

class ViewController: UIViewController {

    private let token: String = "token here"

    @IBAction func verifyMe(_ sender: Any) {
        self.createApplicant(success: { applicantId in
            self.runFlow(withApplicantId: applicantId)
        })
    }
    
    private func createApplicant(success: @escaping (String) -> ()) {
        
        let applicant: Parameters = [
            "first_name": "Theresa",
            "last_name": "May"
        ]
        
        let headers: HTTPHeaders = [
            "Authorization": "Token token=\(self.token)",
            "Accept": "application/json"
        ]
        
        Alamofire.request(
            "https://api.onfido.com/v2/applicants",
            method: .post,
            parameters: applicant,
            encoding: JSONEncoding.default,
            headers: headers).responseJSON { (response: DataResponse<Any>) in
                
                guard response.error == nil else {
                    fatalError("unable to create applicant")
                    return
                }
                
                let response = response.result.value as! [String: Any]
                
                guard response.keys.contains("error") == false else {
                    fatalError("unable to create applicant")
                    return
                }
                
                let applicantId = response["id"] as! String
                success(applicantId)
        }
    }
    
    private func runFlow(withApplicantId applicantId: String) {
        
        let config = try! OnfidoConfig.builder()
            .withToken(self.token)
            .withApplicantId(applicantId)
            .withAppearance(self.getOnfidoAppearance()) // UI customisation
//            .withDocumentStep(ofType: .drivingLicence, andCountryCode: "GBR") // Flow customisation, documents select
            .withDocumentStep()
            .build()
        
        let onfidoFlow = OnfidoFlow(withConfiguration: config)
            .with(responseHandler: { response in
                
                switch response {
                case .success(_):
                    print("success")
                case .error(let error):
                    fatalError("callback error \(error)")
                case .cancel:
                    print("user canceled flow")
                }
            })
        
        do {
            
            let onfidoRun = try onfidoFlow.run()
            onfidoRun.modalPresentationStyle = .formSheet // to present modally
            self.present(onfidoRun, animated: true, completion: {})
            
        } catch let error {
            
            fatalError("unable to run onfido flow \(error)")
        }
    }
    
    private func getOnfidoAppearance() -> Appearance {
        return Appearance(
            primaryColor: UIColor.red,
            primaryTitleColor: UIColor.green,
            primaryBackgroundPressedColor: UIColor.gray,
            secondaryBackgroundPressedColor: UIColor.purple)
    }
}
```
You can find example project here:
[issue-60.zip](https://github.com/onfido/onfido-ios-sdk/files/2673129/issue-60.zip)

Is there any additional information you could provide about your integration?

Anurag

**Comment by tnmendes** [comment]
Hi @anuragajwani,

Thanks for the feedback, I have attached a project that is happening the crash, I think is related with Swinject and the use of Generic UIViewController.

[onfido_bug.zip](https://github.com/onfido/onfido-ios-sdk/files/2679390/onfido_bug.zip)

**Comment by tnmendes** [comment]
Hi @anuragajwani 

I already found the cause of the error, is not related to generic classes, but with hiding the status bar. 

`override public var prefersStatusBarHidden: Bool { return true }`

Is strange that hiding is causing this kind of error with SDK, anyway is fixed on my side.

By the way, I am going to https://onfido.com/dashboard/v2/ then to Checks and final with flag View Test Data to true. I can see the request made using web SDK but I can't see the ones that I did with iOS SDK. It's supposed to be like this?
